tests/test_digest_handler.py

```
@@ -42,290 +65,368 @@ def _make_vault(tmp_path: Path) -> Path:
+    assert "change_note_quality: degraded" in result.body_md
+
+
+def test_audit_event_emitted_on_generate(tmp_path: Path):
+    """``digest_generated`` lands in pipeline.jsonl with input_hash + layer counts."""
+    vault = _make_vault(tmp_path)
+    as_of = datetime.now(timezone.utc) - timedelta(minutes=30)
+    _seed_window_with_signal(vault, as_of=as_of)
+    llm = _FakeLLM("body")
+    handle_digest(_ctx(vault, llm))
+    log_path = vault / "60-Logs" / "pipeline.jsonl"
+    rows = [json.loads(l) for l in log_path.read_text(encoding="utf-8").splitlines() if l]
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)